### ISSUE TITLE
NSXT 3.0.1 Transport Node with VDS 7

### ISSUE THREAD

**Issue by Chadd-C** [vmware/ansible-for-nsxt#311]
I'm not sure this is the right forum, but is there a module that supports configuring transport nodes with VDS 7? Based on the current modules that I have tried, I am only ever able to get a Transport Node (or cluster) to deploy successfully with creating an NVDS.

I am pretty sure that 'host_switch_type' has to be set to 'VDS' in the 'host_switch_spec', but not sure if that is correct.

**Comment by Mybarry** [comment]
That is correct.  You can create a TNP with nsxt_transport_node_profiles and set the host_switch_name to your vSphere VDS name, and set the host_switch_type to "VDS"

**Comment by Chadd-C** [comment]
That is exactly what I've been attempting to run, but I get an error 'Cannot specify HostSwitchMode without specifying TransportZoneEndpoints at HostSwitch <host_switch_id>.'

`"display_name": "TN-Profile01",
            "host_switch_spec": {
                "host_switches": [
                    {
                        "host_switch_id": "50 2f 82 76 b2 48 87 01-2e ab 77 f2 3f 3c 8e 82",
                        "host_switch_mode": "STANDARD",
                        "host_switch_name": "dvSwitch0 (10Gb Nics)",
                        "host_switch_profile_ids": [
                            {
                                "key": "UplinkHostSwitchProfile",
                                "value": "10c0f13d-2aa8-4d5e-9b44-ba5de771956b"
                            }
                        ],
                        "host_switch_type": "VDS",
                        "ip_assignment_spec": {
                            "ip_pool_id": "495c0b44-e78a-4f4b-bf95-e0635a166e8b",
                            "resource_type": "StaticIpPoolSpec"
                        },
                        "pnics": [
                            {
                                "device_name": "Uplink 1",
                                "uplink_name": "uplink-1"
                            },
                            {
                                "device_name": "Uplink 2",
                                "uplink_name": "uplink-2"
                            },
                            {
                                "device_name": "Uplink 3",
                                "uplink_name": "uplink-3"
                            },
                            {
                                "device_name": "Uplink 4",
                                "uplink_name": "uplink-4"
                            }
                        ]
                    }
                ],
                "resource_type": "StandardHostSwitchSpec"
            },
            "hostname": "10.69.33.211",
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "port": 443,
            "resource_type": "TransportNodeProfile",
            "state": "present",
            "transport_zone_endpoints": [
                {
                    "transport_zone_id": "62beb730-ab1c-4451-982a-38607a1b67ae"
                }
            ],
            "username": "admin",
            "validate_certs": false
        }
    },
    "item": {
        "description": "Compute Transport Node Profile",
        "display_name": "TN-Profile01",
        "host_switches": [
            {
                "host_switch_mode": "STANDARD",
                "host_switch_name": "dvSwitch0 (10Gb Nics)",
                "host_switch_profiles": [
                    {
                        "name": "TN-Overlay-Uplink-Profile01",
                        "type": "UplinkHostSwitchProfile"
                    }
                ],
                "host_switch_type": "VDS",
                "ip_assignment_spec": {
                    "ip_pool_name": "s001-TN-TEP-IP-Pool01",
                    "resource_type": "StaticIpPoolSpec"
                },
                "pnics": [
                    {
                        "device_name": "Uplink 1",
                        "uplink_name": "uplink-1"
                    },
                    {
                        "device_name": "Uplink 2",
                        "uplink_name": "uplink-2"
                    },
                    {
                        "device_name": "Uplink 3",
                        "uplink_name": "uplink-3"
                    },
                    {
                        "device_name": "Uplink 4",
                        "uplink_name": "uplink-4"
                    }
                ]
            }
        ],
        "transport_zone_endpoints": [
            {
                "transport_zone_name": "o-test2"
            }
        ]
    },
    "msg": "Failed to add transport node profile. Request body........... 'error_message': 'Cannot specify HostSwitchMode without specifying TransportZoneEndpoints at HostSwitch 50 2f 82 76 b2 48 87 01-2e ab 77 f2 3f 3c 8e 82.'})]."`

**Comment by vgrosdoy** [comment]
How can we specify the vcenter name on which the VDS is hosted ? It is a required field in the GUI, but I do not see the parameter in the nsxt_transport_node_profiles task
```
        host_switches:
        - host_switch_name: "site1_VDS"
          host_switch_mode: "STANDARD"
          host_switch_type: "VDS"
```
Thanks
